### PR TITLE
:man_cook: Make cook routine

### PR DESCRIPTION
**Pull request by Peralban**
To obtain more details see :
- #14

### CODE REVIEW

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
I don't legit know what to say

File: src/kitchen/cook.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "kitchen.hpp"
+#include <string>
+
+Cook::Cook(std::shared_ptr<MessageQueueThread<std::string>> messageQueue) :
+        _messageQueue(messageQueue), _thread(), _pizzaType(Plazza::NONE), _status(WAITING) {}
+
+void Cook::cookRoutine(std::shared_ptr<MessageQueueThread<std::string>> messageQueue)
+{
+    while (true) {
+        int pizzaTypeTmp =  std::stoi(messageQueue->pop());
```
Comment (landryarki) [inline src/kitchen/cook.cpp:18]: you should check if the message queue is empty.

**Review by Thomaltarix — COMMENTED** [review]
There's too many useless pointers, but since we're friday i'll not request it. But check at landry's comments

File: src/kitchen/cook.cpp (hunk 1 above)
Comment (landryarki) [inline src/kitchen/cook.cpp:18]: the format of the message you get from the queue is the following : "t s" with
t = [0 .. 9] * (pizza type)
s = [0 .. 9] * (pizza size)

exemple: "16 8"

you should take that into account when analysing the message from the queue

**Review by landryarki — CHANGES_REQUESTED** [review]
the cook is a little sus

File: src/kitchen/cook.cpp (hunk 1 above)
Comment (Peralban) [reply src/kitchen/cook.cpp:18]: I discuss with thomas about that

Comment (Thomaltarix) [reply src/kitchen/cook.cpp:20]: The landry's message didn't pass but check teams to see its message @Peralban

**Review by Thomaltarix — APPROVED** [review]
The stoi could crash but since it's saturday,  we'll fix it later

**Comment by Peralban** [comment]
> The stoi could crash but since it's saturday, we'll fix it later

The stoi can crash only if the kitchen push shitty string in the queue.

**Review by landryarki — APPROVED** [review]

**Pull request merged**